File: year2/systems-design-implementation/Lab8/SwimProject/SwimModel/src/main/java/services/IService.java
package services;

import domain.Inscriere;
import domain.Participant;
import domain.Proba;
import domain.User;

import java.util.List;

public interface IService {

    void login(User user, Observer client) throws Exception;

    void logout(User user) throws Exception;

    List<Proba> findAllProbe() throws Exception;

    Participant addParticipant(Participant participant) throws Exception;

    Inscriere addInscriere(Inscriere inscriere) throws Exception;

    int nrParticipanti(Proba proba) throws Exception;

    List<Participant> getInscrisi(Proba proba) throws Exception;

    List<Proba> getInscrieri(Participant participant) throws Exception;
}

File: year2/systems-design-implementation/Lab6/SwimModel/Services/IService.cs
using SwimModel.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SwimModel.Services
{
    public interface IService
    {
        void Login(User user, Observer client);

        void Logout(User user);

        IEnumerable<Proba> FindAllProbe();

        Proba FindProba(int id);

        Participant AddParticipant(Participant participant);

        Inscriere AddInscriere(Inscriere inscriere, User user);

        int NrParticipanti(Proba proba);

        IEnumerable<Participant> GetInscrisi(Proba proba);

        IEnumerable<Proba> GetInscrieri(Participant participant);
    }
}
